hal/api/toolchain.h

```diff
@@ -207,6 +207,10 @@
  *  Mark a function declaration as deprecated, if it used then a warning will be
  *  issued by the compiler possibly including the provided message. Note that not
  *  all compilers are able to display the message.
+ *
+ *  @note
```

Comment:
Let's not commit to a date in the documentation like this. Just state it is marked for future removal.